retroshare-gui/src/util/imageutil.cpp

```diff
@@ -137,34 +137,40 @@ bool ImageUtil::optimizeSize(QString &html, const QImage& original, QImage &opti
 	//std::cout << html.toStdString() << std::endl;
 }
 
-int ImageUtil::checkSize(QString &embeddedImage, const QImage &img, int maxBytes)
+bool ImageUtil::optimizeSizeHtml(QString &html, const QImage& original, QImage &optimized, int maxPixels, int maxBytes)
+{
+	QByteArray bytearray;
+	if(maxBytes > 0){
+		maxBytes = maxBytes * 3/4 - 50; //base64 and html stuff
```

Comment:
What happens if someone calls this with maxBytes <= 64 ? I think you should add a test here to make sure this doesn't cause some unwanted behavior.